### PR TITLE
HPCC-14246 Allow DynamicESDL service methods to be written in JAVA

### PR DESCRIPTION
**Pull request by richardkchapman**
Fix some build issues.

Load java plugin dynamically so that jvm is only loaded when the feature is
used.

Signed-off-by: Richard Chapman rchapman@hpccsystems.com

### CODE REVIEW

**Comment by richardkchapman** [comment]
@ghalliday @afishbeck Please review

File: esp/services/esdl_svc_engine/esdl_binding.hpp [hunk 1]
```diff
@@ -96,7 +105,19 @@ class EsdlServiceImpl : public CInterface, implements IEspService
...
+#ifdef LINK_STATICALLY  // May be useful for debugging...
             javaplugin.setown(javaembed::getEmbedContext());
+#else
+            javaPluginDll.setown(createDllEntry("javaembed", false, NULL));
+            if (!javaPluginDll)
+                throw makeStringException(0, "Failed to load javaembed plugin");
+            GetEmbedContextFunction pf = (GetEmbedContextFunction) javaPluginDll->getEntry("getEmbedContextDynamic");
+            if (!pf)
+                throw makeStringException(0, "Failed to load javaembed plugin");
+            javaplugin.set(pf());
```
Comment (ghalliday) [inline esp/services/esdl_svc_engine/esdl_binding.hpp]: should be .setown

File: esp/services/esdl_svc_engine/esdl_binding.hpp [hunk 2]
```diff
@@ -63,7 +64,11 @@ static const char* ESDL_METHOD_HELP="help";
 #define REQ_REF_NUM_NAME    "_req_ref_num"
 #define MCACHE_OBJECT_KEY   "_mcache_object_key_"
 
+typedef IEmbedContext * (* GetEmbedContextFunction)();
```
Comment (ghalliday) [inline esp/services/esdl_svc_engine/esdl_binding.hpp]: Possibly move to hqlplugins.hpp, so that the same mechanism can be used for other language plugins.

**Comment by afishbeck** [comment]
@richardkchapman looks good

**Pull request merged**